### DISCUSSION TITLE
How can I set NODE_ENV for nodejs project when building by Buildpack?

### DISCUSSION THREAD

**Discussion by hofo23** [paketo-buildpacks/feedback#216] [Node.js Team]
Hi, I am struggling to build project by buildpack, I need to overwrite default values by values defined in gitlab pipeline. I see always used only default value "production" set. 

in gitlab pipeline I am using :
AUTO_DEVOPS_BUILD_IMAGE_CNB_BUILDER: "paketobuildpacks/builder:base"

Can you recommend me how can I set it up?   

Status: Downloaded newer image for paketobuildpacks/run:base-cnb
===> ANALYZING
Image with name "tmp-cnb-image" not found
===> DETECTING
6 of 11 buildpacks participating
paketo-buildpacks/ca-certificates       3.6.2
paketo-buildpacks/node-engine           1.6.0
paketo-buildpacks/npm-install           1.1.4
paketo-buildpacks/node-run-script       1.0.10
paketo-buildpacks/npm-start             1.0.11
paketo-buildpacks/environment-variables 4.5.2
===> RESTORING
===> BUILDING
Paketo Buildpack for CA Certificates 3.6.2
  https://github.com/paketo-buildpacks/ca-certificates
  Launch Helper: Contributing to layer
    Creating /layers/paketo-buildpacks_ca-certificates/helper/exec.d/ca-certificates-helper
Paketo Buildpack for Node Engine 1.6.0
  Resolving Node Engine version
    Candidate version sources (in priority order):
                -> ""
      <unknown> -> ""
    Selected Node Engine version (using ): 18.16.1
  Executing build process
    Installing Node Engine 18.16.1
      Completed in 2.349s
  Generating SBOM for /layers/paketo-buildpacks_node-engine/node
      Completed in 0s
  Configuring build environment
    NODE_ENV     -> "production"
    NODE_HOME    -> "/layers/paketo-buildpacks_node-engine/node"
    NODE_OPTIONS -> "--use-openssl-ca"
    NODE_VERBOSE -> "false"
  Configuring launch environment
    NODE_ENV     -> "production"
    NODE_HOME    -> "/layers/paketo-buildpacks_node-engine/node"
    NODE_OPTIONS -> "--use-openssl-ca"
    NODE_VERBOSE -> "false"
    Writing exec.d/0-optimize-memory
      Calculates available memory based on container limits at launch time.

**Comment by robdimsdale** [comment]
cc @paketo-buildpacks/nodejs-maintainers
